cmake/AddSpectreExecutable.cmake

```diff
@@ -33,6 +33,7 @@ function(add_spectre_executable TARGET_NAME)
     RULE_LAUNCH_LINK "${CMAKE_BINARY_DIR}/tmp/WrapExecutableLinker.sh"
     LINK_DEPENDS "${CMAKE_BINARY_DIR}/tmp/WrapExecutableLinker.sh"
     )
+  target_link_options(${TARGET_NAME} PRIVATE "-DEXECUTABLE_NAME=${TARGET_NAME}")
```

Comment:
I'm not sure what to do about the fact that someone can rename the executable on disk. I don't have a change request here, just something worth being aware of. Maybe we should call it something like "EXECUTABLE_LABEL"? I'm not sure, just wanted to bring it up :)

Reply:
Yea I thought about that too, and don't know what to do about it. We may just have to skip the check if the metadata specifies an absolute path.

Reply:
I wonder if what we really "want" are 2 different things:
1. control the executable used for testing
2. check that the underlying executable, whatever it may be, is actually the right one for the input file. That is, we label the executables, but that label doesn't have to be the name. Now, for testing we could just have it be the same.
This could mean executables check that the input file "ExecutableLabel" is the same as the internal label, and then for tests, we just use that as the name in `BUILD/bin`.

Again, not sure, just some musings.

Reply:
Right, I'll revisit this with the scheduling tool.

Reply:
Yea, sounds good. Like I said, I don't know the answer and there's no reason to delay this PR for it. Thanks!